File: 1653-minimum-deletions-to-make-string-balanced/main.cpp
class Solution {
public:
    int minimumDeletions(string s) {

        int b_count = 0, deletions = 0;
        
        for (char chr: s) {
            if (chr == 'b'){ b_count++; }
            else {
                deletions = std::min(deletions + 1, b_count);
            }
        }

        return deletions;
    }
};

File: 1653-minimum-deletions-to-make-string-balanced/main.rs
impl Solution {
    pub fn minimum_deletions(s: String) -> i32 {
        
        let (mut b_count, mut deletions) = (0, 0);

        for chr in s.chars() {
            match chr {
                'b' => b_count += 1,
                _ => deletions = std::cmp::min(deletions + 1, b_count),
            }
        }

        deletions
    }
}
